src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -4,109 +4,151 @@
-                  postHelper.updateInventory(withFunds)
-                    .thenAccept(withInventory -> {
-
-                      logger.info("Successfully Placed Order: " + JsonObject.mapFrom(withCompPO).encodePrettily());
-                      httpClient.closeClient();
-                      javax.ws.rs.core.Response response = PutOrdersByIdResponse.respond204();
-                      AsyncResult<javax.ws.rs.core.Response> result = Future.succeededFuture(response);
-                      asyncResultHandler.handle(result);
-                    })
-                    .exceptionally(this::handleError);
-                })
-                .exceptionally(this::handleError);
-            })
-            .exceptionally(this::handleError)
-        )
-        .exceptionally(this::handleError)
-    ).exceptionally(this::handleError);
+
+    logger.debug("Updating order...");
+    JsonObject purchaseOrder = getJsonOrderWithoutUnusedSubObjects(compPO);
+    operateOnSubObj(HttpMethod.PUT, resourceByIdPath(PURCHASE_ORDER, id), purchaseOrder, httpClient, ctx, okapiHeaders, logger)
+      .thenAccept(v -> {
+
+        logger.info("Applying Funds...");
+        postHelper.applyFunds(compPO)
+          .thenAccept(withFunds -> {
+
+            logger.info("Updating Inventory...");
+            postHelper.updateInventory(withFunds)
+              .thenAccept(withInventory -> {
+
+                logger.info("Successfully Placed Order: " + JsonObject.mapFrom(compPO).encodePrettily());
+                httpClient.closeClient();
+                javax.ws.rs.core.Response response = PutOrdersByIdResponse.respond204();
+                AsyncResult<javax.ws.rs.core.Response> result = Future.succeededFuture(response);
+                asyncResultHandler.handle(result);
+              })
+              .exceptionally(this::handleError);
+          })
+          .exceptionally(this::handleError);
+      })
+      .exceptionally(this::handleError);
+
     return future;
   }
 
-  public Void handleError(Throwable throwable) {
-    final Future<javax.ws.rs.core.Response> result;
-
-    logger.error("Exception while updating orders", throwable.getCause());
+  private CompletableFuture<Void> handlePoLines(CompositePurchaseOrder compOrder, JsonArray poLinesFromStorage) {
+    List<CompletableFuture<?>> futures = new ArrayList<>();
+    List<PoLine> linesToCreate = new ArrayList<>();
+    if (poLinesFromStorage.isEmpty()) {
+      linesToCreate.addAll(compOrder.getPoLines());
+    } else {
+      futures.addAll(filterPoLinesAndUpdateMatching(compOrder, poLinesFromStorage, linesToCreate));
+    }
+    poLinesFromStorage.stream().forEach(poLine ->futures.add(deletePoLine((JsonObject) poLine, httpClient, ctx, okapiHeaders, logger)));
+    boolean updateInventory = compOrder.getWorkflowStatus() == CompositePurchaseOrder.WorkflowStatus.OPEN;
+    linesToCreate.forEach(poLine -> futures.add(new PostOrderLineHelper(httpClient, okapiHeaders, asyncResultHandler, ctx)
```

Comment:
Probably for consistency it makes sense to create instance of the `PostOrderLineHelper` in the constructor like other helpers...